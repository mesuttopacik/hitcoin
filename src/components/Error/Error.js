import { View, Text } from 'react-native'
import React from 'react'
import styles from './Error.styles'


const Error = ({err}) => {
  return (
    <View style = {styles.container}>
        <Text style ={styles.errorMessage} testID='error-message'>{err}</Text>
      <Text>Something went Wrong!  Check your internet connection and try again</Text>
    </View>
  )
}

export default Error