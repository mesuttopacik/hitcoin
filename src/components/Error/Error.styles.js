import { StyleSheet } from 'react-native'

export default StyleSheet.create({
    container:{
        flex:1,
        justifyContent:'center',
        alignItems:'center'
    },
    errorMessage:{
        color:'red',
        fontSize:20,
    }
})