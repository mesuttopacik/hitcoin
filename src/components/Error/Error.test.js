import React from 'react';
import {render} from '@testing-library/react-native';
import Error from './Error';

test('should match with snapshot', () => {
  const comp = render(<Error />);
  expect(comp).toMatchSnapshot();
});

test('should render error correctly', () => { 
    const testErrorMessage = 'an error message'
    const comp = render(<Error err={testErrorMessage}/>)
    const errorMassage = comp.getByTestId('error-message').children[0]
    console.log(errorMassage)
    expect(testErrorMessage).toBe(errorMassage)
 })