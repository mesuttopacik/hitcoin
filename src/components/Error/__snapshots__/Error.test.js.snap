// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "red",
        "fontSize": 20,
      }
    }
    testID="error-message"
  />
  <Text>
    Something went Wrong!  Check your internet connection and try again
  </Text>
</View>
`;
